### ISSUE TITLE
Anchor new scaffolds new program with mismatched Anchor version

### ISSUE THREAD

**Issue by kgilliam125** [coral-xyz/anchor#2358]
When using `anchor new` to scaffold a new program, the `Cargo.toml` file is scaffolded with an Anchor version of 0.23.0 . This is present in Anchor 0.26.0.

## Steps to reproduce
1. Make sure you are on Anchor 0.26.0
2. Create a new anchor workspace with `anchor init`
3. Add a new program to the workspace with `anchor new test`
4. Check the `Cargo.toml` in the newly created program to observe incorrect anchor version.

Note this also causes `anchor build -p test` to fail.

## Expected behavior
The scaffolded Cargo.toml for the new program should match the current Anchor version.

**Comment by Henry-E** [comment]
Looking through the source for `anchor new` the cargo.toml template uses the constant `VERSION` which is defined by an environment variable linked to the CLI somehow, i can't see where the env variable is initially defined. https://github.com/coral-xyz/anchor/blob/27bb6956850477ecbb497dc6c8ff855d97501401/cli/src/lib.rs#L51 

My best guess is that your anchor-cli may not be on 0.26.0 or that the environment variable is manually defined somewhere else that I haven't managed to find yet.

**Comment by Henry-E** [comment]
If you're having trouble with updating anchor versions, you can try uninstalling the npm version and using avm 
https://solana.stackexchange.com/questions/2659/how-to-get-rid-of-an-old-anchor-cli-version
it might also require restarting the terminal as well.

**Comment by kgilliam125** [comment]
Hey @Henry-E thanks for the quick response. I did a little more digging on my end to see what's going on. I installed Anchor on this laptop before avm existed and just patched versions. Switched over to the new method and all seems well. Probably was a dangling env var like you suggested, but glad this isn't an issue!

![image](https://user-images.githubusercontent.com/9299965/211630693-68e71543-b603-4ca2-a313-e74ff164fae1.png)
